src/components/Drawer/PDrawer.vue

```diff
@@ -0,0 +1,175 @@
+<template>
+  <teleport to="body">
```

Comment:
I'm still thinking through some use cases here - I'm thinking to add support for wrapping content (so the drawer takes on an "aside" feel), which would allow the drawer to either overlay content or push it to the side

Reply:
I've decided to defer this functionality until I have a better use case (also this PR is way too large as it is)